wpimath/src/main/java/edu/wpi/first/math/filter/SlewRateLimiter.java

```diff
@@ -14,29 +14,50 @@
  * edu.wpi.first.math.trajectory.TrapezoidProfile} instead.
  */
 public class SlewRateLimiter {
-  private final double m_rateLimit;
+  private final double m_positiveRateLimit;
+  private final double m_negativeRateLimit;
   private double m_prevVal;
   private double m_prevTime;
 
   /**
-   * Creates a new SlewRateLimiter with the given rate limit and initial value.
+   * Creates a new SlewRateLimiter with the given positive and negative rate limits and initial
+   * value.
    *
-   * @param rateLimit The rate-of-change limit, in units per second.
+   * @param positiveRateLimit The rate-of-change limit in the positive direction, in units per
+   *     second. This is expected to be positive.
+   * @param negativeRateLimit The rate-of-change limit in the negative direction, in units per
+   *     second. This is expected to be negative.
```

Comment:
What happens when it's not negative? Should this be a "must"? If so, should this precondition be enforced at runtime?

Reply:
If you had a limit range of, say, 10 to 20, the filter would always slew positive regardless of the input. It's technically correct behavior, but not that useful.